app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,30 @@
+class CommentsController < ApplicationController
+  before_action :user_auth
+
+  def create
+    answer = Answer.find_by(id: comment_params[:answer_id])
+    return render json: { message: 'Answer not found' }, status: :unprocessable_entity if answer.nil?
+
+    comment = Comment.new(comment_params.merge(user_id: @current_user.id))
+    return render json: { message: comment.errors.full_messages }, status: :unprocessable_entity unless comment.save
+
+    render json: comment, status: :created
+  end
+
+  def destroy
+    @comment = Comment.find_by(id: params[:id])
```

Comment:
dont use class instance variable if the scope of the variable is local to the function.